### ISSUE TITLE
Terraform provider version is undefined if none is supplied in cdktf.json

### ISSUE THREAD

**Issue by ansgarm** [hashicorp/terraform-cdk#684]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions
0.3.0

### Affected Resource(s)

azure provider

### Debug Output

```
Error: Invalid version constraint
  on cdk.tf.json line 10, in terraform.required_providers:
  10:       "azurerm": {
  11:         "version": "undefined",
  12:         "source": "azurerm"
  13:       }
This string does not use correct version constraint syntax.
non-zero exit code 1
```

### Expected Behavior

It should pick up the latest one

### Actual Behavior

It produces invalid configuration

### Steps to Reproduce

`cdktf.json` with no versions
```
{
  "language": "typescript",
  "app": "npm run --silent compile && node main.js",
  "terraformProviders": ["azurerm", "digitalocean/digitalocean"],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
"allowSepCharsInLogicalIds": "true"
  }
}
```

**Comment by Zubnix** [comment]
same issue with helm provider

**Comment by 0hyunU** [comment]
it should be written the version on cdktf.json in terraformProviders. 
Maybe you can check the correct provider version at each provider example directory

After inserting the version, download dependencies by 'cdktf get' command

{
  "language": "typescript",
  "app": "npm run --silent compile && node main.js",
  "terraformProviders": [
    "azurerm@~> 2.0.0"
  ]
}

**Comment by djsd123** [comment]
Same with the aws provider.  As @ansgarm alluded to. It ideally, should offer a similar user experience as terraform(non cdk). 

For Posterity `"terraformProviders": ["aws@~>3.44.0"]`

**Comment by PacoVK** [comment]
This issue links to https://github.com/hashicorp/terraform-cdk/issues/228#issuecomment-859090427 

TL;DR `cdktf get` should have an option simular to `terraform init -upgrade` to reload (and update) provider

**Comment by djsd123** [comment]
I just dug into this a little bit to see how it could be implemented and why it was not implemented already.  It appears that it is not as straightforward as I initially thought.  

`terraform` uses logic that is not immediately obvious to me to determine what version of the provider is the latest as opposed to having a queryable endpoint like for example `https://registry.terraform.io/v1/providers/hashicorp/aws/latest`.  

I think `terraform` determines from `https://registry.terraform.io/v1/providers/hashicorp/aws/versions` what version it should fetch.  However, my conclusion comes from only 15 minutes of looking. 🤷‍♂️

**Comment by PacoVK** [comment]
Yes, however there are many reasons why you would pin the version hence refering to latest is not a good idea. 
The current implementation does only support a `terraform init` or `terrafrom init -input=false` i suggest two possible options to go along: 
* force version constraint in `cdktf.json` and throw an error in an earlier stage when validating `cdktf.json`
* use a mechanism like Terraform generating a `terraform.lock.hcl` to "pin" the version

The low hanging fruit is option 1 , however for both it makes sense to additionally introduce a flag to easily upgrade the provider, eg `cdktf get -upgrade`
